flexmeasures/cli/utils.py

```diff
@@ -316,3 +316,29 @@ def validate_color_hex(ctx, param, value):
         return value
     else:
         raise click.BadParameter(f"{param.name} must be a valid hex color code.")
+
+
+def validate_url(ctx, param, value):
+    """
+    Validates that a given value is a valid URL format using regex.
+
```

Comment:
Assuming you didn't write the regex from scratch, please cite its source in the docstring in accordance with the licence set by the publisher. For instance, the StackOverflow licence demands a citation.

Reply:
You're right, I didn't write most of these myself. After creating the initial validation, I was experimenting with different regex patterns and got some helpful suggestions from GitHub Copilot. Although I didn't need the majority of them, I'll take those out and keep only the important one.